packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -442,13 +493,242 @@ export class Dialog
+    this.transitionEl.classList.toggle(CSS.openingActive, value);
     onToggleOpenCloseComponent(this);
   }
 
+  private async reflowInteraction(): Promise<void> {
+    const { interaction } = this;
+
+    if (!interaction) {
+      return;
+    }
+
+    await interaction.reflow({
+      name: "drag",
+    });
+
+    await interaction.reflow({
+      name: "resize",
+    });
+  }
+
+  private getDOMRect(): DOMRect {
+    return this.transitionEl?.getBoundingClientRect() ?? new DOMRect();
+  }
+
+  private handleKeyDown = (event: KeyboardEvent): void => {
+    const { key, shiftKey, defaultPrevented } = event;
+    const { dragEnabled, resizable, resizePosition, dragPosition } = this;
+
+    const keys = ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"];
+
+    if (defaultPrevented || !keys.includes(key)) {
+      return;
+    }
+
+    switch (key) {
+      case "ArrowUp":
+        if (shiftKey && resizable) {
+          this.setSize({ size: this.getDOMRect().height + dialogStep, type: "blockSize" });
+          resizePosition.bottom += dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        } else if (dragEnabled) {
+          dragPosition.y -= dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        }
+        break;
+      case "ArrowDown":
+        if (shiftKey && resizable) {
+          this.setSize({ size: this.getDOMRect().height - dialogStep, type: "blockSize" });
+          resizePosition.bottom -= dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        } else if (dragEnabled) {
+          dragPosition.y += dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        }
+        break;
+      case "ArrowLeft":
+        if (shiftKey && resizable) {
+          this.setSize({ size: this.getDOMRect().width - dialogStep, type: "inlineSize" });
+          resizePosition.right -= dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        } else if (dragEnabled) {
+          dragPosition.x -= dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        }
+        break;
+      case "ArrowRight":
+        if (shiftKey && resizable) {
+          this.setSize({ size: this.getDOMRect().width + dialogStep, type: "inlineSize" });
+          resizePosition.right += dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        } else if (dragEnabled) {
+          dragPosition.x += dialogStep;
+          this.setTransform();
+          this.reflowInteraction();
+          event.preventDefault();
+        }
+        break;
+    }
+  };
+
+  private setTransform(): void {
```

Comment:
Would `updateTransform` be more appropriate? A set function without arguments seems unusual to me.